proposed/extended-coding-style-guide.md

```diff
@@ -82,9 +82,9 @@ each word is capitalised including the very first letter.
 
 ### 2.2 Files
 
-All PHP files MUST use the Unix LF (linefeed) line ending.
+All PHP files MUST use the Unix LF (linefeed) line ending, applying to every line in the file.
 
-All PHP files MUST end with a single line, containing only a single newline (LF) character.
+All PHP files MUST end with a non-blank line, terminated with a single LF like all other lines in the file.
```

Comment:
I tend to agree. My original approach was based on:

* Defining a <line> (zero or more non-<newline> characters followed/terminated by a <newline>)
* Stating a file consists of zero or more <line>s
* Defining a blank <line>
* Stating that a blank line cannot go on the end of the file.

We can show what that looks like in various editors in the metadoc.